### PR TITLE
Add support for ADL devices - SNDW and I2S

### PR DESCRIPTION
**Pull request by sathyap-chrome**
Commits add support for ADL match table and adds support for
(a) 2 customer boards with I2S codecs
(b) ADL P RVP with SNDW codec

### CODE REVIEW

**Comment by sofci** [comment]
Can one of the admins verify this patch?
> reply **test this please** to run this test once.

**Comment by sathyap-chrome** [comment]
will push update with checkpatch fixes

**Review by kv2019i — DISMISSED** [review]
Excellent, thank you @sathyap-chrome and @Vamshigopal ! I'll let @plbossart and @bardliao look at the sof_sdw changes, but rest looks good.

File: sound/soc/sof/intel/pci-tgl.c
Comment (plbossart) [inline sound/soc/sof/intel/pci-tgl.c:102]: typo in commit message: segregate

File: sound/soc/intel/boards/sof_sdw.c
Comment (plbossart) [inline sound/soc/intel/boards/sof_sdw.c]: can we use the same match criteria as other platforms

````
			DMI_MATCH(DMI_SYS_VENDOR, "Intel Corporation"),
			DMI_MATCH(DMI_PRODUCT_NAME, "Ice Lake Client"),
````

**Review by plbossart — CHANGES_REQUESTED** [review]
looks mostly good, I just have 2 minor comments below.

File: sound/soc/intel/boards/sof_sdw.c
Comment (sathyap-chrome) [reply sound/soc/intel/boards/sof_sdw.c]: The dmidecode on adlrvp didnt show vendor as INTEL. hence used the family. will talk to BIOS team and see if that can be fixed soon.

File: sound/soc/sof/intel/pci-tgl.c
Comment (sathyap-chrome) [reply sound/soc/sof/intel/pci-tgl.c:102]: Ack

File: sound/soc/sof/intel/pci-tgl.c
Comment (bardliao) [inline sound/soc/sof/intel/pci-tgl.c:95]: Should we add a comment like ` /* Alderlake */` in `const struct sof_intel_dsp_desc tgl_chip_info = {}` if it is for ADL, too?

File: sound/soc/intel/boards/sof_sdw.c
Comment (sathyap-chrome) [reply sound/soc/intel/boards/sof_sdw.c]: this will be done by coreboot team.I'll verify the coreboot image and update V2 tomorrow

File: sound/soc/sof/intel/pci-tgl.c
Comment (sathyap-chrome) [reply sound/soc/sof/intel/pci-tgl.c:95]: sure. will do

File: sound/soc/intel/boards/sof_sdw.c
Comment (sathyap-chrome) [reply sound/soc/intel/boards/sof_sdw.c]: @plbossart with coreboot being used, the DMI match for vendor wont be possible as suggested. The corresponding coreboot.org conversation for reference :
https://review.coreboot.org/c/coreboot/+/51408/comment/3a850ca7_f58f7f79

Hence i think we should go ahead with current FAMILY string or i can change to product name ONLY. 

Let me know your thoughts and i can publish v2 as per that.

Comment (plbossart) [reply sound/soc/intel/boards/sof_sdw.c]: sorry, I don't get the point at all. There is no reason why an Intel board should be named differently because of coreboot/UEFI changes.

@kv2019i what are the other RVPs using with the UEFI BIOS?

File: sound/soc/sof/intel/pci-tgl.c
Comment (kv2019i) [reply sound/soc/sof/intel/pci-tgl.c:102]: @sathyap-chrome ping on this? if you haven't done the change yet, I'd change this from "segragate" to "separate".

Comment (sathyap-chrome) [reply sound/soc/sof/intel/pci-tgl.c:102]: its done :) will post along with v2

File: sound/soc/intel/boards/sof_sdw.c
Comment (sathyap-chrome) [reply sound/soc/intel/boards/sof_sdw.c]: @plbossart found the fix. now we can use Intel and adlrvp as match. posting v2 now

**Review by plbossart — APPROVED** [review]
LGTM thanks @sathyap-chrome

**Review by bardliao — APPROVED** [review]
LGTM. Thanks @sathyap-chrome

**Comment by sathyap-chrome** [comment]
@kv2019i can you also approve. Also can this be merged ?
@plbossart when is the tentative timeline this lands in alsa mailing list ? since customer is waiting, i can provide the date.

**Comment by plbossart** [comment]
SOFCI TEST

**Comment by plbossart** [comment]
@fredoh9 @marc-hb is Jenkins in the weeds?

**Comment by plbossart** [comment]
> @kv2019i can you also approve. Also can this be merged ?
> @plbossart when is the tentative timeline this lands in alsa mailing list ? since customer is waiting, i can provide the date.

I'll merge when CI works.
Tentative upstream is next week.

File: sound/soc/intel/boards/sof_sdw.c
Comment (fredoh9) [reply sound/soc/intel/boards/sof_sdw.c]: For ADL UEFI BIOS,
```
			DMI_MATCH(DMI_SYS_VENDOR, "Intel Corporation"),
			DMI_MATCH(DMI_PRODUCT_NAME, "Alder Lake Client Platform"),
```

**Comment by fredoh9** [comment]
Jenkins works fine, sometime being throttled. good to go.

File: sound/soc/intel/boards/sof_sdw.c
Comment (plbossart) [reply sound/soc/intel/boards/sof_sdw.c]: @sathyap-chrome Can you please revisit why we need to have different DMI names for CoreBoot? that's not good. Fix Coreboot!

Comment (sathyap-chrome) [reply sound/soc/intel/boards/sof_sdw.c]: @fredoh9 can you try my patch and confirm it works with UEFI. let me talk to coreboot engineer on why there is a difference.

**Review by kv2019i — APPROVED** [review]

**Comment by kv2019i** [comment]
Thanks @sathyap-chrome and reviewers. Merging this now.

File: sound/soc/intel/boards/sof_sdw.c
Comment (plbossart) [inline sound/soc/intel/boards/sof_sdw.c:194]: merged too quickly @kv2019i: this doesn't work for a UEFI BIOS.

Comment (kv2019i) [reply sound/soc/intel/boards/sof_sdw.c:194]: Oops, sorry @plbossart , I read the comments in wrong order... the comment on this was more recent than your approval. Let's do a fixup for this before sending upstream. Should we have another match case for UEFI case? @sathyap-chrome and @libinyang  ?

Comment (sathyap-chrome) [reply sound/soc/intel/boards/sof_sdw.c:194]: @plbossart  
I requested coreboot to fix the match to be same as UEFI BIOS. once the fix is merged, i can send the fixup. maybe a day or 2.

Comment (plbossart) [reply sound/soc/intel/boards/sof_sdw.c:194]: no worries, let's do a fixup

Comment (libinyang) [reply sound/soc/intel/boards/sof_sdw.c:194]: @kv2019i , We need the quirk for UEFI. The DMI info is like what @fredoh9 provided.

Comment (plbossart) [reply sound/soc/intel/boards/sof_sdw.c:194]: @sathyap-chrome ping. we need the fix to upstream this series. We only have 2-3 days left before the next cycle. Thanks for prioritizing this.

Comment (sathyap-chrome) [reply sound/soc/intel/boards/sof_sdw.c:194]: @plbossart  will share the new patch today itself

**Pull request merged**
